actor/v7action/service_instance_sharing_test.go

```diff
@@ -0,0 +1,171 @@
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"some-service-instance-warning"},
+					ccerror.ServiceInstanceNotFoundError{Name: serviceInstanceName, SpaceGUID: targetedSpaceGUID},
+				)
+			})
+
+			It("returns an actor error and warnings", func() {
+				Expect(fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceCallCount()).To(Equal(1))
+				Expect(fakeCloudControllerClient.GetOrganizationsCallCount()).To(Equal(0))
+				Expect(fakeCloudControllerClient.GetSpacesCallCount()).To(Equal(0))
+
+				Expect(executionError).To(MatchError(actionerror.ServiceInstanceNotFoundError{Name: serviceInstanceName}))
+				Expect(warnings).To(ConsistOf("some-service-instance-warning"))
+			})
+		})
+
+		When("the service instance is retrieved", func() {
+			BeforeEach(func() {
+				fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceReturns(
+					resources.ServiceInstance{Name: serviceInstanceName},
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"some-service-instance-warning-1"},
+					nil,
+				)
+			})
+
+			When("the space to share to cannot be found", func() {
+				BeforeEach(func() {
+					fakeCloudControllerClient.GetSpacesReturns(
+						[]resources.Space{},
+						ccv3.IncludedResources{},
+						ccv3.Warnings{"some-space-warnings"},
+						nil,
+					)
+				})
+
+				It("returns an actor error and warnings", func() {
+					Expect(fakeCloudControllerClient.GetSpacesCallCount()).To(Equal(1))
+					Expect(fakeCloudControllerClient.GetSpacesArgsForCall(0)).To(Equal(
+						[]ccv3.Query{
+							{
+								Key:    ccv3.NameFilter,
+								Values: []string{shareToSpaceName},
+							},
+							{
+								Key:    ccv3.OrganizationGUIDFilter,
+								Values: []string{targetedOrgGUID},
+							},
+						}))
+
+					Expect(executionError).To(MatchError(actionerror.SpaceNotFoundError{Name: shareToSpaceName}))
+					Expect(warnings).To(ConsistOf("some-service-instance-warning-1", "some-space-warnings"))
+				})
+
+				When("the org to share to is also specified", func() {
+					const shareToOrgGUID = "share-to-org-guid"
+
+					BeforeEach(func() {
+						serviceInstanceSharingParams.OrgName = types.NewOptionalString(shareToOrgName)
+
+						fakeCloudControllerClient.GetOrganizationsReturns(
+							[]resources.Organization{{GUID: shareToOrgGUID}},
+							ccv3.Warnings{"some-org-warning"},
+							nil,
+						)
+					})
+
+					It("searches for the space in the specified org", func() {
+						Expect(fakeCloudControllerClient.GetOrganizationsCallCount()).To(Equal(1))
+						Expect(fakeCloudControllerClient.GetSpacesCallCount()).To(Equal(1))
+						Expect(fakeCloudControllerClient.GetSpacesArgsForCall(0)).To(Equal(
+							[]ccv3.Query{
+								{
+									Key:    ccv3.NameFilter,
+									Values: []string{shareToSpaceName},
+								},
+								{
+									Key:    ccv3.OrganizationGUIDFilter,
+									Values: []string{shareToOrgGUID},
+								},
+							}))
+					})
+
+					It("returns an actor error and warnings", func() {
```

Comment:
I missed a "when the space cannot be found in the specified org". Reading this spec, we end up with:

When the org to share to is also specified
  It searches for the space in the specified org
  It returns an actor error and warnings

Not a big deal though

Reply:
My idea was 
- space cannot be found
  - it errors in default case (targeted org)
  - when org is also specified
   - It searches for the space in the specified org
   - It returns an actor error and warnings


Might be worth revisiting this after all tests have been written to see the flow of al the test cases